lib/collection/src/collection/distance_matrix.rs

```diff
@@ -0,0 +1,142 @@
+use std::collections::HashSet;
+use std::time::Duration;
+
+use rand::prelude::SliceRandom;
+use segment::data_types::vectors::{
+    NamedMultiDenseVector, NamedVectorStruct, VectorStructInternal,
+};
+use segment::types::{Filter, HasIdCondition, PointIdType, ScoredPoint, WithVector};
+
+use crate::collection::Collection;
+use crate::operations::query_enum::QueryEnum;
+use crate::operations::shard_selector_internal::ShardSelectorInternal;
+use crate::operations::types::{
+    CollectionResult, CoreSearchRequest, CoreSearchRequestBatch, PointRequestInternal,
+};
+
+#[derive(Debug, Default)]
+struct DistanceMatrixResult {
+    sample_ids: Vec<PointIdType>,   // sampled point ids
+    nearest: Vec<Vec<ScoredPoint>>, // nearest points for each sampled point
+}
+
+impl Collection {
+    async fn distance_matrix(
+        &self,
+        sample_count: usize,
+        limit: usize,
```

Comment:
1. What is the difference between `sample_count` and `limit`?

2. nit: maybe `sample_size` is a better name?

Reply:
1. sample count = number of points to select randomly
    limit = number of neighbors per sample to compute
    
2. maybe :+1: